villain/test/test-runner.rkt

```diff
@@ -651,7 +651,20 @@
   (check-equal? (run-with-file (λ (file)
                                   `(let ((port (open-input-file ,file)))
                                      (begin (close-input-port port) (close-input-port port)))) "")
-                (void)))
+                (void))
```

Comment:
This seems like not nearly enough tests for this language feature.  There should be something testing functions being passed in as arguments and returned as results.  There should also be uses of lambda with fixed and variable arity.

Reply:
Yes, will add more tests. Thanks!